[Settings] Change icons and text

Jay-o-Way opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

1. Updated some icons as from #10331 and (personal suggestions in) #18866.
   - General Page > "Run as admin" probably needs some tuning with the text and link.  _(did it in VS Code, please do check)_
2. Removed description which is incorrect for Win.11, as mentioned in #19590
3. Tweak text for Camera overlay image #18573
4. Added `IsEnabled` on Infobar on ScG page #16885
5. Tweak colors to adress #16261 _(might not be perfect yet, please do check)_
6. Use "primary/seconday", not "left/right" mouse buttons #19742
+ Feedback from pr

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [ ] **Closes:** #xxx
- [x] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [ ] **Tests:** Added/updated and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed

htcfreek commented:
@jay-o-way
There is a warning box in PT Run's plugin manager that has an incorrect message. Can you update it?

**String id**: `Run_NotAllowedKeywordWarning.Title`
**Current text is something like:** `This activation command is used for other plugins too.`
**Better message:** `This activation command is forbidden or reserved for the use in other plugins.`

Jay-o-Way commented:
> The message is shown on forbidden keywords and not on multiple used keywords.

Sure, changing a bit of text is easy. But how do I trigger this message again?

Jay-o-Way commented:
@htcfreek how do I trigger that message, again?

htcfreek commented:
> @htcfreek how do I trigger that message, again?

Using keywords like `\` or `\\`

htcfreek commented on src/settings-ui/Settings.UI/Views/ShortcutGuidePage.xaml:
```diff
@@ -68,6 +68,7 @@
                         IsTabStop="True"
                         IsOpen="{x:Bind Mode=OneWay, Path=ViewModel.UseLegacyPressWinKeyBehavior}"
                         IsClosable="False"
+                        IsEnabled="{x:Bind Mode=OneWay, Path=ViewModel.IsEnabled}"
```
Isn't this fixed. I think I opened an issue in the past.

Jay-o-Way replied:
Not yet
![image](https://user-images.githubusercontent.com/65828559/180643105-f3ed1621-7607-4a66-9845-04c4d08600c0.png)

Jay-o-Way commented:
I would like to know what 1 and 5 look like

Jay-o-Way commented:
@niels9001 the `x:Uid="ShortcutGuide_Appearance_Behavior` is (re)used in a number of places. For a user it doesn't matter a thing, but would it be good to at least generalize the name?

Jay-o-Way commented:
And yes, I realize the "left/right" commits increase the number of files. Reason is that, when I thought of creating another branch just for that, the last commit wasn't a clean one: it contained changes in files unrelated to this subject. Trying to sort that out - without starting all over - was too difficult. Or I wasn't patient enough.

Jay-o-Way commented:
Anybody?

niels9001 commented:
@jay-o-way Are the left/rouse mouse button updates intentional? I understand updating labels, but why updating variable names in the core code (and were you able to test those changes)?

Jay-o-Way commented:
> @jay-o-way Are the left/rouse mouse button updates intentional? I understand updating labels, but why updating variable names in the core code (and were you able to test those changes)?

Well, it started with those (two) user facing words, then the xaml bindings. After that I just wanted to keep going.
Haven't been able to really test it. My computer is not that fast, and the bot doesn't give errors any more. That's why a check from you guys is always welcome.

niels9001 commented:
> > @jay-o-way Are the left/rouse mouse button updates intentional? I understand updating labels, but why updating variable names in the core code (and were you able to test those changes)?
> 
> Well, it started with those (two) user facing words, then the xaml bindings. After that I just wanted to keep going.
> Haven't been able to really test it. My computer is not that fast, and the bot doesn't give errors any more. That's why a check from you guys is always welcome.

Since this PR is about updating user-facing text labels and icons in Settings, I'd stick to that. Now you're touching multiple projects in a single PR (I'd not touch those var names anyways, since they are not user facing..).

If you revert those I'll do a pass on the icons and text labels and approve 😄

niels9001 commented on src/settings-ui/Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -606,9 +606,6 @@
   <data name="PowerRename_Toggle_ContextMenu.Header" xml:space="preserve">
     <value>Show PowerRename in</value>
   </data>
-  <data name="PowerRename_Toggle_ContextMenu.Description" xml:space="preserve">
```
Isn't this still needed for Windows 10 devices?

Jay-o-Way replied:
"needed"is a difficult word. Up to now, it's also shown in Win.11 and that's just wrong. And (to quote some unnamed persons) "PowerToys is intented for power-users", right? I'd expect power users know how to show the extended menu.

niels9001 replied:
I'd leave this in for now as it's a regression.. We can open a new issue and make a helper class that hides this message when  on W11. Would be useful for other things (in the future as well)?

Jay-o-Way replied:
That helper is a fine idea too - we already have one. (for example to hide a "rounded corners" settings for FZ.)

niels9001 commented on src/settings-ui/Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1256,7 +1255,7 @@ Made with 💗 by Microsoft and the PowerToys community.</value>
   </data>
   <data name="Run_NotAllowedActionKeyword.Title" xml:space="preserve">
-    <value>This activation command is already in use by another plugin.</value>
+    <value>This activation command is forbidden or reserved for the use in other plugins.</value>
```
This is more verbose.. and not not super clear to me, IMO.

Jay-o-Way replied:
Was suggested a few commenys up here. Must have had a reason. After what condition is this text actually visible?

htcfreek replied:
This change can be reverted as this text will be removed/replaced in my PR. See https://github.com/microsoft/PowerToys/pull/19593#issuecomment-1209180786.

Jay-o-Way replied:
Is there an _e.t.a._ for that pr? I would be unhappy if I revert this now, and my pr would make it in the next version and yours would (for whatever reason) not... Seems safer to me to let you just overwrite it.

niels9001 commented on src/settings-ui/Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -128,7 +128,7 @@
     <value>Video Conference Mute is a quick and easy way to do a global "mute" of both your microphone and webcam. Disabling this module or closing PowerToys will unmute the microphone and camera.</value>
   </data>
   <data name="VideoConference_CameraAndMicrophoneMuteHotkeyControl_Header.Header" xml:space="preserve">
-    <value>Mute camera &amp; microphone</value>
```
Specific reason why moving from '&' to 'and'? Windows 11 Settings seem to favor & over 'and' for most cases (albeit not everywhere).. also more condensed?

Jay-o-Way replied:
This is a case of personal opinion; didn't check guidance or something like that. Although it doesn't really matter a ything, as the following images show:
![image](https://user-images.githubusercontent.com/65828559/183284616-4651e473-02b8-442f-b88a-d2cf8a685eb4.png)![image](https://user-images.githubusercontent.com/65828559/183284633-561c9ddc-b011-4487-9c91-620f0e576fcd.png)

niels9001 replied:
<img width="272" alt="image" src="https://user-images.githubusercontent.com/9866362/183284923-f9e3fee9-c589-4db9-ab05-68f748a7d336.png">

& seems a bit more formal, and seems to be used in W11 Settings for header for Settings groups (although it's a bit all over the place). Nice Sunday read :): https://medium.com/bpxl-craft/analyzing-the-ampersand-d0dc0daf81f3

Jay-o-Way replied:
Nice try. What about https://docs.microsoft.com/style-guide/a-z-word-list-term-collections/a/ampersand ? (MS Docs) 😉

Jay-o-Way replied:
I guess you do have a point. That means the display is not how it would be wanted.

Jay-o-Way replied:
Maybe I should just use a literal `&` then...

niels9001 commented on src/settings-ui/Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1247,7 +1246,7 @@ Made with 💗 by Microsoft and the PowerToys community.</value>
   </data>
   <data name="Run_NotAccessibleWarning.Title" xml:space="preserve">
-    <value>Please define an activation command or allow this plugin for the global results to use it.</value>
+    <value>Please define an activation command or allow this plugin to be used in the global results.</value>
```
Like this change :)!

niels9001 commented on src/settings-ui/Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1354,10 +1353,10 @@ Made with 💗 by Microsoft and the PowerToys community.</value>
   </data>
   <data name="Oobe_ShortcutGuide.Description" xml:space="preserve">
-    <value>Shortcut Guide presents the user with a listing of available shortcuts for the current state of the desktop.</value>
+    <value>Shortcut Guide presents you a listing of available shortcuts for the current state of the desktop.</value>
```
Nice! This is inline with the Windows Writing Style :).. we should use 'you' everywhere instead of 'user'

niels9001 commented on src/settings-ui/Settings.UI/Views/KeyboardManagerPage.xaml:
```diff
@@ -182,13 +182,10 @@
+                                                <SolidColorBrush Color="{ThemeResource SystemAccentColor}"
                                                                  Opacity="0.3"/>
                                             </Border.Background>
-                                            <TextBlock Text="{x:Bind TargetApp}"
```
The foreground must be set to White, not it will turn black in light mode and white in dark mode... We need to come up with a better design overall for the Apps token.

Jay-o-Way replied:
> The foreground must be set to White, not it will turn black in light mode and white in dark mode...

If whe use white, then it'll be impossible to read when the accent color is getting light. I thought about this, and in my head it should work, given that the background color still has transparency.

> We need to come up with a better design overall for the Apps token.

Do agree with that. Maybe a grid? Better to do this together with #2027

niels9001 replied:
Yeah we need some sort of table design.. I'll check what the latest is on Token design guidelines.

Jay-o-Way replied:
How does it look on your end? (Dark and light theme)

niels9001 replied:
![image](https://user-images.githubusercontent.com/9866362/183861465-dc25e1c7-7725-44f3-9055-4a5493886764.png)

SystemAccentColor should only be used for text labels (when inside of controls), not directly as a backdrop. This can result in a11y issues:

![image](https://user-images.githubusercontent.com/9866362/183861663-0fbc56f1-d4e9-4170-8beb-948656733b8d.png)

I'd be in favor in reverting this for now and creating a new issue and do a separate PR for redesigning the entire thing.

Jay-o-Way replied:
Hm, those two screenshots look pretty clean. I don't see the problem in these?

Jay-o-Way replied:
> SystemAccentColor should only be used for text labels (when inside of controls), not directly as a backdrop. This can result in a11y issues:

1. You use SystemAccentColor as the backdrop for the buttons in the ItemsControl. So...?
2. I know it can lead to [issues](https://github.com/microsoft/PowerToys/issues/16261), that's the reason I'm trying this :)

niels9001 replied:
The contrast between the background of the token and the row background will probably not meet the right standards. Either way, we need to update the whole listview and rethink where we place what and how 😄.

Jay-o-Way commented:
(Woops, my bad)

niels9001 commented on src/settings-ui/Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1900,11 +1897,11 @@ From there, simply click on one of the supported files in the File Explorer and
   </data>
-  <data name="MouseUtils_MouseHighlighter_LeftButtonClickColor.Header" xml:space="preserve">
-    <value>Left button highlight color</value>
+  <data name="MouseUtils_MouseHighlighter_PrimaryButtonClickColor.Header" xml:space="preserve">
```
Just as a double check: MouseHighlighter works as expected when turning left/right mouse buttons around, right :)?

Jay-o-Way replied:
Up to now, the words are simply wrong. I think this is something you can easily test/experience yourself. :)

niels9001 commented on src/modules/powerrename/PowerRenameUILib/MainWindow.xaml:
```diff
@@ -279,9 +279,9 @@
+                    <ToggleButton x:Name="toggleButton_includeFiles" Content="&#xE7C3;" MinHeight="0" Height="31" IsChecked="True" FontFamily="{ThemeResource SymbolThemeFontFamily}" x:Uid="ToggleButton_IncludeFiles" Style="{StaticResource CustomToggleButtonStyle}" />
                     <ToggleButton x:Name="toggleButton_includeFolders" Content="&#xE8B7;" MinHeight="0" Height="31" IsChecked="True" FontFamily="{ThemeResource SymbolThemeFontFamily}" x:Uid="ToggleButton_IncludeFolders" Style="{StaticResource CustomToggleButtonStyle}" />
-                    <ToggleButton x:Name="toggleButton_includeSubfolders" Content="&#xE12F;" MinHeight="0" Height="31" IsChecked="True" FontFamily="{ThemeResource SymbolThemeFontFamily}" x:Uid="ToggleButton_IncludeSubFolders" Style="{StaticResource CustomToggleButtonStyle}" />
+                    <ToggleButton x:Name="toggleButton_includeSubfolders" Content="&#xED41;" MinHeight="0" Height="31" IsChecked="True" FontFamily="{ThemeResource SymbolThemeFontFamily}" x:Uid="ToggleButton_IncludeSubFolders" Style="{StaticResource CustomToggleButtonStyle}" />
```
This looks like vertical folder? Not sure if it's better than the original folder, which atleast showed some sort of nesting/subfolders?

Jay-o-Way replied:
E12F is not only deprecated, but also a right-aligned bulleted list? We've had a discusson about using check boxes earlier and I still stand behind that.

niels9001 commented on src/settings-ui/Settings.UI/Views/GeneralPage.xaml:
```diff
@@ -172,12 +172,11 @@
                     <controls:SettingExpander IsExpanded="True">
                         <controls:SettingExpander.Header>
                             <controls:Setting x:Uid="Admin_Mode" Icon="&#xE7EF;" Description="{Binding Mode=OneWay, Path=RunningAsText}" Style="{StaticResource ExpanderHeaderSettingStyle}">
+                                <HyperlinkButton NavigateUri="https://aka.ms/powertoysDetectedElevatedHelp" x:Uid="GeneralPage_ToggleSwitch_AlwaysRunElevated_Link"/>
```
![image](https://user-images.githubusercontent.com/9866362/183862859-78d5aafc-fe72-4b1c-ac1c-48eec914dc9f.png)

Where did this go? You have a Description set here already. The HyperLink needs to be in the Description.

niels9001 replied:
But what happens then with the {Binding Mode=OneWay, Path=RunningAsText} 🤔?

Jay-o-Way replied:
Yeah, this is the one I actually need the feedback on. My idea is visible in my issue.

jaimecbernardo replied:
@Jay-o-Way , Reviewing this and the issue. The idea is to put the link close to the "Running as user" / "Running as admin", right? I agree that can make sense, since Administrator mode is just spelled out twice already.
What's blocking here?

Jay-o-Way replied:
The strings don't show as I want and the link doesn't seem to appear at all.

jaimecbernardo replied:
Hi, but the idea is to move "running as user" to the top and have the hyperlink as description, right?

jaimecbernardo replied:
Something like this?
![image](https://user-images.githubusercontent.com/26118718/185442950-2dabe051-e1fc-405d-a78f-9af51f08276a.png)

jaimecbernardo replied:
What needs to be done is to get this is putting the HyperLink button inside a `<controls:Setting.Description>`, set the Header instead of the description to "RunningAsText" and change the Id of the Setting so that it doesn't get the Header from Resources.

![image](https://user-images.githubusercontent.com/26118718/185443555-1971b279-035d-4115-8b85-184615fb3edf.png)

Hope this helps.

jaimecbernardo replied:
@Jay-o-Way , hope this helps

Jay-o-Way replied:
Yes! You got it! ❤

Jay-o-Way commented on src/settings-ui/Settings.UI/Views/ColorPickerPage.xaml:
```diff
@@ -127,11 +127,7 @@
-                                                        <FontIcon Glyph="&#xF0AE;" />
-                                                    </MenuFlyoutItem.Icon>
-                                                </MenuFlyoutItem>
+                                                <MenuFlyoutItem IsEnabled="{x:Bind CanMoveDown}" Icon="Down" x:Uid="MoveDown" Click="ReorderButtonDown_Click"/>
```
That's strange. Does it say why? I just used the same style as "Up" already was...

Jay-o-Way replied:
Also #19080

niels9001 replied:
It's because the "Down" icon is not recognized. It's one of the reasons we use glyph codes everywhere, the Symbol names (as listed on the Docs) do not always seem to work.

Jay-o-Way replied:
Fine, that'll let me fix _this_ inconsistency while I'm at it...
![image](https://user-images.githubusercontent.com/65828559/185793899-669ddcc9-c101-42db-a4a1-f74db73da7da.png)
 Can I use `<MenuFlyoutItem Icon=E74B"...` here or will that fail too?

niels9001 replied:
Use <FontIcon ../> just to be sure.

Jay-o-Way commented on src/settings-ui/Settings.UI/Views/FancyZonesPage.xaml:
```diff
@@ -65,7 +65,7 @@
                     <controls:SettingExpander IsExpanded="True">
                         <controls:SettingExpander.Header>
-                            <controls:Setting x:Uid="FancyZones_ZoneBehavior_GroupSettings" Icon="&#xE620;" Style="{StaticResource ExpanderHeaderSettingStyle}" />
+                            <controls:Setting x:Uid="FancyZones_ZoneBehavior_GroupSettings" Style="{StaticResource ExpanderHeaderSettingStyle}" />
```
I understand what you mean here, but this "mis-alignment" is also in different places, so _if_ it really is a problem, then we should fix it _everywhere_. Is it possible to adapt the custom control to (p)reserve that space if no icon is set?
![image](https://user-images.githubusercontent.com/65828559/183872196-d1c04989-2c08-48b2-bef2-58f7535fc05c.png)

Jay-o-Way replied:
Conclusion: I'm leaving it for now. The custom control is off-topic and (I think) something you can easily change.

niels9001 replied:
I meant inconsistent across the FancyZones settings for window / zones

Jay-o-Way commented on src/settings-ui/Settings.UI/Views/FancyZonesPage.xaml:
```diff
@@ -98,7 +98,7 @@
                         <controls:SettingExpander.Header>
 
-                            <controls:Setting x:Uid="FancyZones_Zone_Appearance" Icon="&#xE790;" Style="{StaticResource ExpanderHeaderSettingStyle}" >
+                            <controls:Setting x:Uid="FancyZones_Zone_Appearance" Icon="&#xEB3C;" Style="{StaticResource ExpanderHeaderSettingStyle}" >
```
Yes there is. I've mentioned it in my issue :)

Jay-o-Way replied:
@niels9001 you okay with that?

niels9001 replied:
It does not literally desceibe 'appearance'. But it it is often used across Windows (and other OSes to describe changing visual preferences).

Jay-o-Way commented on src/settings-ui/Settings.UI/Views/ImageResizerPage.xaml:
```diff
@@ -34,7 +33,7 @@
 
                 <controls:SettingsGroup x:Uid="ImageResizer_CustomSizes" IsEnabled="{x:Bind Mode=OneWay, Path=ViewModel.IsEnabled}">
-                    <controls:Setting x:Uid="ImageResizer_Presets" Icon="&#xE2B2;">
+                    <controls:Setting x:Uid="ImageResizer_Presets" Icon="&#xE792;">
```
This is the one I could find that, for me, has the most to do with "Saving" + the ability to change.

Jay-o-Way replied:
If you have a better suggestion, let me know

Jay-o-Way commented on src/settings-ui/Settings.UI/Views/FancyZonesPage.xaml:
```diff
@@ -169,7 +169,7 @@
                     <controls:SettingExpander IsExpanded="True">
                         <controls:SettingExpander.Header>
-                            <controls:Setting x:Uid="FancyZones_WindowBehavior_GroupSettings" Icon="&#xE737;" Style="{StaticResource ExpanderHeaderSettingStyle}" />
+                            <controls:Setting x:Uid="FancyZones_WindowBehavior_GroupSettings" Style="{StaticResource ExpanderHeaderSettingStyle}" />
```
Same answer

Jay-o-Way commented on src/settings-ui/Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -690,7 +687,7 @@
   </data>
   <data name="PowerRename_Toggle_AutoComplete.Header" xml:space="preserve">
-    <value>Enable auto-complete for the search and replace fields</value>
+    <value>Enable auto-complete for the search &amp; replace fields</value>
```
I know, but it was like this before. (can be seen in early commit) I believe xml/res files don't like single `&` chars.
But the thing is: this is eventually translated into _words_ which you can see in screenshots in the convo between Niels and me.

htcfreek replied:
sure. but this resource is a checkbox name or description. so it is a full sentences and not a header. there we should keep words

Jay-o-Way replied:
Niels said the exact opposite. 😟

Jay-o-Way replied:
Ugh. I need to find a MS doc for this...

niels9001 replied:
I agree with @htcfreek. I'd personally keep ampersands for headers (you want those to be short and to the point) while individual settings can have longer labels?

htcfreek commented on src/settings-ui/Settings.UI/Views/ShortcutGuidePage.xaml:
```diff
@@ -68,6 +68,7 @@
                         IsTabStop="True"
                         IsOpen="{x:Bind Mode=OneWay, Path=ViewModel.UseLegacyPressWinKeyBehavior}"
                         IsClosable="False"
+                        IsEnabled="{x:Bind Mode=OneWay, Path=ViewModel.IsEnabled}"
```
`IsEnabled` doesn't work. You have to use `IsOpen`

Jay-o-Way replied:
Uhm, I don't think so? There's already an `IsOpen` for this, which mean the same as _visibility". The problem is that when you turn off the entire module, this infobar won't change...

htcfreek replied:
But it doesn't work. I tried `IsEnabled` in my PR and it doesn't work. @niels9001 what do you think?

niels9001 replied:
@htcfreek @jay-o-way So there's no visual difference between Enabled/Disabled? Hmm, might be something to flag in the WinUI repo.

As a workaround now I'd set the Visibility or IsOpen instead?

htcfreek replied:
@niels9001
Do you like to open an issue in WinUI repo? I agree that this looks like a bug in WinUI.

Jay-o-Way replied:
@niels9001

- can you explain the difference between `muxc:InfoBar` and `<InfoBar>`? According to [this page](https://docs.microsoft.com/nl-nl/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.infobar?view=windows-app-sdk-1.1#:~:text=Gets%20or%20sets%20a%20value%20indicating%20whether%20the%20user%20can%20interact%20with%20the%20control.) the **IsEnabled** property _should_ be possible, am I right?
- Like I already said - and you all can see - the is already an **IsOpen** property and it has a different purpose.

niels9001 replied:
> @niels9001
> 
> - can you explain the difference between `muxc:InfoBar` and `<InfoBar>`? According to [this page](https://docs.microsoft.com/nl-nl/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.infobar?view=windows-app-sdk-1.1#:~:text=Gets%20or%20sets%20a%20value%20indicating%20whether%20the%20user%20can%20interact%20with%20the%20control.) the **IsEnabled** property _should_ be possible, am I right?
> - Like I already said - and you all can see - the is already an **IsOpen** property and it has a different purpose.

We can remove the muxc tag: this was required for WinUI 2.x (in XAML Islands), but since we moved to WAS this can be removed and we can use e.g. <InfoBar../>

Yes, I am aware that there is an IsEnabled property and that it works (the InfoBar.Content gets disabled right?) but that doesn't mean that there is a Disabled state implemented as well that would update the style so it 'looks' disabled.

Apparently that is not the case, so we need to come up with a fallback.. leaving everything as is or hiding it?
I'd say this Disabled state would be nice to have for this control so that's why we could flag it in the WinUI repo, who owns the the control.

niels9001 replied:
https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/InfoBar/InfoBar.xaml

Looking at the XAML, there is no Disabled state :(. We could add our IsEnabledTextBlock in the Header/Title and Message - that should work. Wrapping the InfoBar in a ContentControl might work (but doubt it tbh).

Jay-o-Way replied:
Other (muxc)InfoBar controls do set visual styles, but this single one is unique in that it has no content. I think it _should_ inherit from `SettingsGroup`, or we need to give it content.

niels9001 replied:
Yesh adding to the SettingsGroup might work.. but let's leave this out of scope for this PR now so we get this in?

Jay-o-Way commented:
I'm close to replacing my sluggish desktop with an all-in-one pc soon. Just waiting until I have both a new CPU and an SSD. And then I have to install literally everything. I'm hoping, by the time I'm done, I can do more with Visual Studio so I can actually test more myself. Until then: I still have a number of unanswered questions in this PR. I hope this can be included in the next release 🤞🏻

jaimecbernardo commented:
Hi @Jay-o-Way ,
Could you please point me at what's still unanswered to expedite? ;) 
Thank you!

niels9001 reviewed (changes requested):
@jaimecbernardo This looks good to go, except for the MenuFlyout change.

Jay-o-Way commented:
1. For the Up and Down icons in the list: "Can I use `<MenuFlyoutItem Icon="E74B"... /> here to remove <MenuFlyoutItem.Icon> ... </>`, or will that fail too?"
2. Apparently, the **IsEnabled** property doesn't seem to work on the `<muxc:InfoBar x:Uid="ShortcutGuide_PressWinKeyWarning"` on the VCM settings page. This _might_ be a WinUI bug. Other muxc:infobars do change their visual style, but they are all constructed differently. Need a work-around for this one. To repeat myself for clarity: the **IsOpen** property has a different purpose and thus is **not** an alternative.

To Niels - _not_ blocking:

- That Windows11 converter/helper is a fine idea too - we already have one. Did you miss that?
- I changed icon E790 "Color" to EB3C "Design" in a few instances. I don't see a "no-go", so I suppose that's good?
- I changed the icon for ImageResizer_Presets → same question.
- We've had a discusson about using check boxes in PowerRename UI earlier and I still stand behind that.

Jay-o-Way commented:
> except for the MenuFlyout change.

@niels9001 Suggestions for actual improvement would be very welcome please

niels9001 commented:
> 1. For the Up and Down icons in the list: "Can I use `<MenuFlyoutItem Icon="E74B"... /> here to remove <MenuFlyoutItem.Icon> ... </>`, or will that fail too?"
> 2. Apparently, the **IsEnabled** property doesn't seem to work on the `<muxc:InfoBar x:Uid="ShortcutGuide_PressWinKeyWarning"` on the VCM settings page. This _might_ be a WinUI bug. Other muxc:infobars do change their visual style, but they are all constructed differently. Need a work-around for this one. To repeat myself for clarity: the **IsOpen** property has a different purpose and thus is **not** an alternative.
> 
> To Niels - _not_ blocking:
> 
> - That Windows11 converter/helper is a fine idea too - we already have one. Did you miss that?
> - I changed icon E790 "Color" to EB3C "Design" in a few instances. I don't see a "no-go", so I suppose that's good?
> - I changed the icon for ImageResizer_Presets → same question.
> - We've had a discusson about using check boxes in PowerRename UI earlier and I still stand behind that.

- Fine with the icon changes.
- Just leave the MenuFlyoutItem code as is and change the glyph code for 'down'.
- We can look at the InfoBar issues in a seperate PR - there's a tracking issue right?
- CheckBoxes for PowerRename is not relevant to this PR? So would move that discussion to a dedicated issue. I do find the subfiles/folders to be a regression to the icon we have now. We can have a look at creating custom icons if necessary later?

Jay-o-Way commented:
Anything keeping this from being merged now?

niels9001 commented:
> Anything keeping this from being merged now?

I can tsst it later this week and will merge if everything works as expected 👍👍

Jay-o-Way commented:
Ping?

crutkas commented:
/azp run

niels9001 commented on src/settings-ui/Settings.UI/Views/ColorPickerPage.xaml:
```diff
@@ -17,133 +17,133 @@
         <StackPanel Orientation="Vertical"
```
@jay-o-way This page crashes with a XAML related error. For you as well? Due to the many formatting changes it's hard to check what was changed.

niels9001 commented on src/settings-ui/Settings.UI/Views/ColorPickerPage.xaml:
```diff
@@ -17,133 +17,133 @@
+                                        VerticalAlignment="Center"
+                                        Grid.RowSpan="2"
+                                        x:Uid="More_Options_Button"
+                                        Content="&#xEDE3;">
```
![image](https://user-images.githubusercontent.com/9866362/188415121-5858cbc5-637f-4786-8ba4-9fc90f41af9c.png)

We had the '...' before right? This is the Xbox controller menu button..

Jay-o-Way replied:
True.

niels9001 commented on src/settings-ui/Settings.UI/Views/ColorPickerPage.xaml:
```diff
@@ -17,133 +17,133 @@
+                                        Content="&#xEDE3;">
+                                    <Button.Flyout>
+                                        <MenuFlyout>
+                                            <MenuFlyoutItem IsEnabled="{x:Bind CanMoveUp}" Icon="e74a" x:Uid="MoveUp" Click="ReorderButtonUp_Click"/>
```
This crashes the app.. Remove the Icon property and add:

  <MenuFlyoutItem.Icon>
                                                    <FontIcon Glyph="&#xe74a;" />
                                                </MenuFlyoutItem.Icon>

Jay-o-Way replied:
Yeah, it's annoying how `Icon` doesn't seem to accept the hex code and `Icon="Down"` doesn't work either 😒
Fixed both.

niels9001 reviewed (changes requested):
@Jay-o-Way Almost there! One weird icon on the ColorPicker page (see comment) and incorrect setting of the MenuFlyoutItem icon. Happy to make the changes for you?

Jay-o-Way reviewed (commented):
fixed

niels9001 reviewed (approved):
LGTM! Thanks!

@jaimecbernardo This can be merged once CI passes!

jaimecbernardo commented:
Thank you! We've already got our Release Candidates, though, so it's going in for 0.63.

htcfreek commented:
@Jay-o-Way , @niels9001 , @jaimecbernardo 
Before merging this please validate that all icons are available on all supported os versions including Windows 10.

Jay-o-Way commented:
@htcfreek all come from https://docs.microsoft.com/windows/apps/design/style/segoe-fluent-icons-font so it's good.

crutkas commented:
in the future for stuff like changing icons, in the PR, would be super powerful to see before / after.  Goes from quick glance to having to compile

Jay-o-Way commented:
> in the future for stuff like changing icons, in the PR, would be super powerful to see before / after.  Goes from quick glance to having to compile

Can do. I had an rather old and very slow desktop. Now I have an all-in-one with SSD.

SteinwaySons commented:
@Jay-o-Way May I ask? I wanna know the question I raised, ["Meaning-unknown Chinese characters in PowerToys UI"](https://github.com/microsoft/PowerToys/issues/20598), had been taken into consideration?

Jay-o-Way commented:
> @Jay-o-Way May I ask? I wanna know the question I raised, ["Meaning-unknown Chinese characters in PowerToys UI"](https://github.com/microsoft/PowerToys/issues/20598), had been taken into consideration?

There is a deprecated icon code used for this icon. These still show in many (non-asian?) languages, therefore it wasn't obvious. It is changed in this pr.

htcfreek commented:
> > @Jay-o-Way May I ask? I wanna know the question I raised, ["Meaning-unknown Chinese characters in PowerToys UI"](https://github.com/microsoft/PowerToys/issues/20598), had been taken into consideration?
> 
> There is a deprecated icon code used for this icon. These still show in many (non-asian?) languages, therefore it wasn't obvious. It is changed in this pr.

wht font do we use? static or `symbolfont` brush?

Jay-o-Way commented:
@htcfreek it's not about the font but the (some deprecated) icon codes: https://learn.microsoft.com/nl-nl/windows/apps/design/style/segoe-fluent-icons-font#icon-list

htcfreek commented:
> @htcfreek it's not about the font but the (some deprecated) icon codes: https://learn.microsoft.com/windows/apps/design/style/segoe-fluent-icons-font#icon-list

But why it's only on the Chinese system?

SteinwaySons commented:
> @Jay-o-Way May I ask? I wanna know the question I raised, ["Meaning-unknown Chinese characters in PowerToys UI"](https://github.com/microsoft/PowerToys/issues/20598), had been taken into consideration?

@Jay-o-Way I just find out why my PT meets trouble in display icon. **Because of _font set_ missing!** Someone gave us a solution on this [link](https://github.com/microsoft/PowerToys/issues/18055). Corresponding [MS docs link](https://learn.microsoft.com/en-us/windows/apps/design/style/segoe-fluent-icons-font#how-do-i-get-this-font) is here. 
I copy the reply here.
> Seems it's caused by the lack of segoe fluent icons, which is not included in Windows 10 by default. Installing [the font](https://aka.ms/SegoeFluentIcons) solved this issue on my machine. [Reference](https://docs.microsoft.com/en-us/windows/apps/design/style/segoe-fluent-icons-font#how-do-i-get-this-font)
PS: this bug just appeared several versions ago, by when I've never installed this font

Here is the snapshot.
![image](https://user-images.githubusercontent.com/37266092/191224956-60d1bd31-ce51-4b93-a659-af58ffe6ec69.png)

So, after I [download Segoe Fluent Icons font](https://aka.ms/SegoeFluentIcons) and install this font set, UI error is gone! 
![image](https://user-images.githubusercontent.com/37266092/191224354-b272beaf-4402-413b-9e68-caf901d6018d.png)
![image](https://user-images.githubusercontent.com/37266092/191224376-a2bbfb86-8f08-4dee-ae7f-f26174c2055f.png)
![image](https://user-images.githubusercontent.com/37266092/191225534-65af131e-dfe3-43cb-9ab2-c1bdb311cdf0.png)

It's work for me!

Jay-o-Way commented:
CC: @niels9001

SteinwaySons commented:
@Jay-o-Way Finally, please accept my apology for misunderstanding. Hope PT will have a good future!

jaimecbernardo reviewed (approved):
LGTM! Thank you for the contribution! ;) Sorry for the delay in merging.
